packages/renderers-js/e2e/system/test/transferSol.test.ts

```diff
@@ -46,6 +51,54 @@ test('it transfers SOL from one account to another', async (t) => {
   t.is(await getBalance(client, destination), lamports(1_000_000_000n));
 });
 
+test('it fails if the source account does not have enough SOLs', async (t) => {
+  // Given a source account with 1 SOL and a destination account with no SOL.
+  const client = createDefaultSolanaClient();
+  const source = await generateKeyPairSignerWithSol(client, 1_000_000_000n);
+  const destination = (await generateKeyPairSigner()).address;
+
+  // When the source account tries to transfer 2 SOLs to the destination account.
+  const transferSol = getTransferSolInstruction({
+    source,
+    destination,
+    amount: 2_000_000_000,
+  });
+  const txMessage = pipe(await createDefaultTransaction(client, source), (tx) =>
+    appendTransactionMessageInstruction(transferSol, tx)
+  );
+  const promise = signAndSendTransaction(client, txMessage);
+
+  // Then we expect the transaction to fail and to have the correct error code.
+  const error = await t.throwsAsync(promise);
+  if (
+    isSolanaError(
+      error,
+      SOLANA_ERROR__JSON_RPC__SERVER_ERROR_SEND_TRANSACTION_PREFLIGHT_FAILURE
+    )
+  ) {
+    if (isSystemError(error.cause, txMessage)) {
+      error satisfies SolanaError;
```

Comment:
Was this meant to be something like this?

```suggestion
    error satisfies SolanaError<typeof SOLANA_ERROR__JSON_RPC__SERVER_ERROR_SEND_TRANSACTION_PREFLIGHT_FAILURE>;
    if (isSystemError(error.cause, txMessage)) {
        error.cause satisfies SolanaError<typeof SOLANA_ERROR__INSTRUCTION_ERROR__CUSTOM> & {
            context: { code: SystemError };
        };
```

Reply:
Yeah sorry my type checks were not quite as thorough haha. I've updated them. ✅